src/current.rs

```diff
@@ -34,23 +39,74 @@ pub fn clock_for(task_id: TaskId) -> VectorClock {
     ExecutionState::with(|state| state.get_clock(task_id).clone())
 }
 
+/// Apply the given function to the Labels for the specified task
+pub fn with_labels_for_task<F, T>(task_id: TaskId, f: F) -> T
+where
+    F: FnOnce(&mut Labels) -> T,
+{
+    LABELS.with(|cell| {
+        let mut map = cell.borrow_mut();
+        let m = map.entry(task_id).or_default();
+        f(m)
+    })
+}
+
+/// Get a label of the given type for the specified task, if any
+#[inline]
+pub fn get_label_for_task<T: Clone + std::fmt::Debug + Send + Sync + 'static>(task_id: TaskId) -> Option<T> {
+    with_labels_for_task(task_id, |labels| labels.get().cloned())
+}
+
+/// Add the given label to the specified task, returning the old label for the type, if any
+#[inline]
+pub fn set_label_for_task<T: Clone + std::fmt::Debug + Send + Sync + 'static>(task_id: TaskId, value: T) -> Option<T> {
+    with_labels_for_task(task_id, |labels| labels.insert(value))
+}
+
+/// Remove a label of the given type for the specified task, returning the old label for the type, if any
+#[inline]
+pub fn remove_label_for_task<T: Clone + std::fmt::Debug + Send + Sync + 'static>(task_id: TaskId) -> Option<T> {
```

Comment:
Nit: I think these should have the `std::fmt::` part removed.

Reply:
Agreed, and fixed.